splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -1045,8 +1045,19 @@ public boolean hasNext() {
                                                 ObjectInputStream ois = new ObjectInputStream(bais);
                                                 // compose the entry for a given column
                                                 ExecRow statsRow = StatisticsAdmin.generateRowFromStats(conglomId, "-All-", columnId, (ColumnStatisticsImpl) ois.readObject());
-                                                dataDictionary.addColumnStatistics(statsRow, tc);
-                                                bais.close();
+                                                try {
+                                                    dataDictionary.addColumnStatistics(statsRow, tc);
+                                                } catch (StandardException e) {
+                                                    // DB-9890 Skip a column if its statistics object doesn't fit into HBase cell.
+                                                    if (e.getCause().getMessage().contains("KeyValue size too large")) {
+                                                        SpliceLogUtils.warn(LOG, "Statistics object of [ConglomID=%d, ColumnID=%d] exceeds max KeyValue size. Try increase hbase.client.keyvalue.maxsize.",
+                                                                conglomId, columnId);
```

Comment:
Is it possible to return this warning as part of the analyze output in addition to the warning in the log? so that user is aware that some column stats are skipped.

Reply:
I was thinking of something similar but not sure what's the correct way of doing it. It seems the standard way is to throw an exception and set the severity to warning. But we cannot throw here since we want to continue. Building the warning message into the result table would misuse the a char column, unless we add an extra column to the result table schema and set the message there. Not sure we should do that, either.

Do you have any suggestions?

Reply:
I'm thinking of adding an extra column in the analyze output to indicate whether some column stats were not saved due to space limitation. Thanks!

Reply:
Got it, will add another commit.

Reply:
Done.